src/extension.js

```diff
@@ -410,17 +319,55 @@ const ServiceIndicator = GObject.registerClass({
         this.settings.run_dispose();
 
         // Destroy the PanelMenu.SystemIndicator actors
-        this._item.destroy();
         this.menu.destroy();
 
-        delete AggregateMenu._gsconnect;
+        delete QuickSettingsMenu._gsconnect;
         super.destroy();
     }
 });
 
+const FeatureIndicator = GObject.registerClass(
+class FeatureIndicator extends QuickSettings.SystemIndicator {
+    _init() {
+        super._init();
 
-var serviceIndicator = null;
+        // GSettings
+        this.settings = new Gio.Settings({
+            settings_schema: Config.GSCHEMA.lookup(
+                'org.gnome.Shell.Extensions.GSConnect',
+                null
+            ),
+            path: '/org/gnome/shell/extensions/gsconnect/',
+        });
+
+        // Create the icon for the indicator
+        this._indicator = this._addIndicator();
+        this._indicator.icon_name = 'org.gnome.Shell.Extensions.GSConnect-symbolic';
+        // Hide the indicator by default
+        this._indicator.visible = false;
+
+        // Create the toggle menu and associate it with the indicator
+        this.quickSettingsItems.push(new ServiceIndicator());
```

Comment:
```suggestion
        this.quickSettingsItems.push(new ServiceToggle());
```

Renaming for consistency (see other suggestions).